### PR TITLE
Rules for deploying DEB packages (fix #24)

### LINKED ISSUE #24: Create `deploy_deb` and `deploy_rpm` which deploy already built DEB and RPM package s
A user should be able to deploy DEB and RPM using new rule `deploy_deb` and `deploy_rpm`. The rules take in already built DEB and RPM packages which you can generate using the `distribution_deb` and `distribution_rpm` rules.

To illustrate how people should use `deploy_deb` and `deploy_rpm`:
```
distribution_deb(
  name = "distribution-deb",
  ...
)
deploy_deb(
  name = "deploy-deb",
  distribution = "distribution-deb"
)

distribution_rpm(
  name = "distribution-deb",
  ...
)
deploy_rpm(
  name = "deploy-deb",
  distribution = "distribution-deb"
)
```

This is how a user can perform the deployment:
```
bazel run //:deploy-deb 
bazel run //:deploy-rpm
```

You should be able to specify to what server you want to deploy to by configuring the URL in `deployment.properties`:
```
deb.repository-url.pypi=...
deb.repository-url.test=...
rpm.repository-url.pypi=...
rpm.repository-url.test=...

```

### PR DESCRIPTION
**Pull request by vmax**
Adds `deploy_deb` rule.
Sample usage:
```
load("@graknlabs_rules_deployment//deb/deployment:rules.bzl", "deploy_deb")
…
deploy_deb(
    name = "deploy-deb",
    target = ":distribution-deb",
    deployment_properties = "//:deployment.properties",
)
```

### CODE REVIEW

File: deb/templates/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+if [[ $# -ne 3 ]]; then
+    echo "Should pass <test> <deb-username> <deb-password> as arguments"
+    exit 1
+fi
+
+DEB_REPO_TYPE="$1"
+DEB_USERNAME="$2"
+DEB_PASSWORD="$3"
+
+if [[ "$DEB_REPO_TYPE" != "test" ]]; then
```
Comment (lolski) [inline deb/templates/deploy.sh]: Let's add
```
TODO: add support deploying to a non-test repository
```

for this one and `deploy_rpm`

File: deb/templates/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    exit 1
+fi
+
+# create a temporary file for preprocessing deployment.properties
+DEPLOYMENT_PROPERTIES_STRIPPED_FILE=$(mktemp)
+# awk in the next line strips out empty and comment lines
+awk '!/^#/ && /./' deployment.properties > ${DEPLOYMENT_PROPERTIES_STRIPPED_FILE}
+DEB_URL=$(grep "deb.repository-url.$DEB_REPO_TYPE" ${DEPLOYMENT_PROPERTIES_STRIPPED_FILE} | cut -d '=' -f 2)
+
+http_status_code_from_upload=$(curl --silent --output /dev/stderr --write-out "%{http_code}" -X POST -u $DEB_USERNAME:$DEB_PASSWORD -H "Content-Type: multipart/form-data" --data-binary "@package.deb" $DEB_URL)
```
Comment (lolski) [inline deb/templates/deploy.sh]: This line is too long, I would suggest splitting it like this:

```
http_status_code_from_upload=$(curl --silent --output /dev/stderr --write-out "%{http_code}" -X POST \
    -u $DEB_USERNAME:$DEB_PASSWORD -H "Content-Type: multipart/form-data" --data-binary "@package.deb" $DEB_URL)
```

File: deb/templates/deploy.sh (hunk 1 above)
Comment (vmax) [reply deb/templates/deploy.sh]: @lolski I don't think it should be in code. The reason I think so is that I don't know whether mainline repos use the similar procedure ("upload *.deb file by curl") and therefore, the `TODO` item won't be actionable. we have graknlabs/deployment#29 to remind us.

Comment (lolski) [reply deb/templates/deploy.sh]: I see, let's keep it as is for now then.

**Pull request merged**